rskj-core/src/main/java/co/rsk/peg/LegacyErpFederation.java

```diff
@@ -75,13 +76,26 @@ public Script getStandardRedeemScript() {
     }
 
     private void validateRedeemScript() {
+        Script redeemScript = this.getRedeemScript();
+
         if (activations.isActive(ConsensusRule.RSKIP293) &&
-            this.getRedeemScript().equals(new Script(LEGACY_ERP_TESTNET_REDEEM_SCRIPT_BYTES))) {
+            redeemScript.equals(new Script(LEGACY_ERP_TESTNET_REDEEM_SCRIPT_BYTES))) {
 
             String message = "Unable to create ERP Federation. The obtained redeem script matches the one hardcoded for testnet. "
                 + "This would cause bitcoinj-thin to identify it as invalid";
             logger.debug("[validateRedeemScript] {}", message);
-            throw new FederationCreationException(message);
+            throw new FederationCreationException(message, HARDCODED_LEGACY_ERP_TESTNET_REDEEM_SCRIPT);
+        }
+
+        // we need to check if every stack inside the scriptsig has a valid size.
```

Comment:
Below validation is checking the size of the whole scriptsig, not chunks by chunks.

Reply:
actually is just checking the redeem script size since is the only stack element that could exceed the limit

Reply:
```
// this scriptSig contains the signatures, the redeem script and some other bytes
// so it is ok to just check the redeem script size
```